### ISSUE TITLE
Adding alias option on column definition

### ISSUE THREAD

**Issue by thibault-lr** [typeorm/typeorm#4032]
**Issue type:**

[ ] question
[ ] bug report
[X ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ X] `mongodb`
[ X] `mssql`
[ X] `mysql` / `mariadb`
[ X] `oracle`
[ X] `postgres`
[ X] `cockroachdb`
[ X] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[ X] `latest`
[X ] `@next`
[ ] `0.x.x` (or put your version here)

It would be amazing if we could add an "alias" option on the column. 
For example an Entity like this : 

`
@Entity()
export class User {

  @PrimaryGeneratedColumn()
  id: number;

  @Column({
    type: "text"})
  department: string;

  @Column("text")
  localization: string;

}
`

`
getConnection().getRespoitory(User).find()
`

Generates the following statement : 
`SELECT "User"."id" AS "User_id", "User"."department" AS "User_department", "User"."localization" AS "User_localization" FROM "user" "User"`

What about adding an option to columns like for example  : 
`
  @Column({
    type: "text",
    alias: "dep"
})
  department: string;
`

So the statement will be like this : 
`SELECT "User"."id" AS "User_id", "User"."department" AS "dep", "User"."localization" AS "User_localization" FROM "user" "User"`

The only workaround I found is to change the class name of the entity or to use the query builder

**Comment by Kononnable** [comment]
I don't really see what will we gain by that. At the end of the day you still use javascript objects and shouldn't care much of generated sql as long as it gives you results you want and is performant.
Readable sql is good in case of debbuging - but adding column aliases just for that is just a bit overkill.

**Comment by vlapo** [comment]
I agree with @Kononnable. In simple cases it is nice to have. In more complex cases you will use query builder and define your aliases there.

**Comment by Iruora** [comment]
@Kononnable , I think it will be useful for oracle databases related projects as oracle defines a maximum length of 32 chars and when at least two columns names contains 30 chars for each, the generated SQL query will cause an error :  Error: ORA-00918: column ambiguously defined

**Comment by kyle-mccarthy** [comment]
I am running into an issue with the query builder generating ambiguously named columns columns too. I am actually trying to join two entities School and SchoolDistrict. The school has a district_id column and then SchoolDistrict has an id column. When joined the query builder selects `school.district_id AS school_district_id` and selects `school_district.id AS school_district_id`...

**Comment by HexMox** [comment]
Sometimes the table of database is already existed. we cannot query by defining an Entity now because of incorrect column name.


This doesn't work.
```js
const repository = connection.getRepository(Entity);
const qb = repository.createQueryBuilder(Entity.name);
EntityMetaData.columns.map((x) => {
      qb.addSelect(`${Entity.name}.${x.propertyName}`, x.databaseName);
});
const data = await qb.where(`${Entity.name}.id = :id`, { id }).getOne();
```

**Comment by HexMox** [comment]
This works.

```ts
import { Driver } from 'typeorm';
import { DriverUtils } from 'typeorm/driver/DriverUtils';
import StringUtils from 'typeorm/util/StringUtils';

// @ts-ignore
DriverUtils.buildColumnAlias = function({ maxAliasLength }: Driver, alias: string, column: string) {
  const columnAliasName = column;
  if (maxAliasLength && maxAliasLength > 0 && columnAliasName.length > maxAliasLength) {
    return StringUtils.hash(columnAliasName, { length: maxAliasLength });
  }
  return columnAliasName;
};
```

**Comment by bombillazo** [comment]
As @HexMox mentioned, sometimes the database table already exists, or the column could be defined by another model and you need to reference the column in another model (which could happen in inherited tables).

Would be nice to be able to reference a column on a class field without trying to recreate the column.

**Comment by samonkoba** [comment]
Another use case that would really benefit from alias is with error: `ER_NONUNIQ_TABLE: Not unique table/alias` on the @Authorize directive.  Sample below:

`@Authorize({
    authorize: (context: UserContext) => ({
        or: [
            {apartment: {**owner**: {agents: {agentId: {eq: context.req.user.id}}}}},
            {**owner**: {agents: {agentId: {eq: context.req.user.id}}}},
        ]
    } as unknown)
})
`

Note: Owner in this case can exist in multiple resources but it fails since i think it uses same alias
